### PR TITLE
Add Attach Database dialog

### PR DESCRIPTION
**Pull request by corivera**
Currently waiting on merging these changes in STS: https://github.com/microsoft/sqltoolsservice/pull/2193

Closes https://github.com/microsoft/azuredatastudio/issues/23249

AttachDatabase

### CODE REVIEW

**Comment by erinstellato-ms** [comment]
@corivera thoughts on this order of info?  Open to discussion here...

image

**Comment by corivera** [comment]
With that ordering, it seems unclear to me whether the Attach As and Owner parts apply to the Database or the associated files. In SSMS the editable fields are in the database table itself, so putting controls underneath the table is the next best thing we can manage.

**Review by erinstellato-ms — APPROVED** [review]

File: extensions/mssql/src/objectManagement/ui/attachDatabaseDialog.ts [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+import { DefaultMinTableRowCount, DialogButton, getTableHeight } from '../../ui/dialogBase';
+import path = require('path');
+import { getErrorMessage } from '../../utils';
+
+export class AttachDatabaseDialog extends ObjectManagementDialogBase<Database, DatabaseViewInfo> {
+	private _databasesToAttach: DatabaseFileData[] = [];
+	private _databasesTable: azdata.TableComponent;
+	private _associatedFilesTable: azdata.TableComponent;
+	private _databaseFiles: string[][] = [];
+	private readonly _fileFilters: azdata.window.FileFilters[] = [{ label: loc.DatabaseFilesFilterLabel, filters: ['*.mdf'] }];
```
Comment (ssreerama) [inline extensions/mssql/src/objectManagement/ui/attachDatabaseDialog.ts:22]: question: Attaching .mdf file is enough to attach a database? will it have the references of its related .ndf and .ldf files?

**Review by ssreerama — APPROVED** [review]

File: extensions/mssql/src/objectManagement/ui/attachDatabaseDialog.ts (hunk 1 above)
Comment (corivera) [reply extensions/mssql/src/objectManagement/ui/attachDatabaseDialog.ts:22]: We get all of the associated files in a separate query. This is the same file browser as SSMS.

**Pull request merged**